abs_clamp/abs_clamp.tex

```diff
@@ -0,0 +1,213 @@
+
+Secondly, we need to show that the vector entries belong to the interval \texttt{[L, U]}. This follows from the definition of \texttt{function} in line \ref{line:fn}. According to line \ref{line:fn} in the pseudocode, there are 3 possible cases to consider:
+\begin{enumerate}
+    \item \texttt{x} $>$ \texttt{U}: then $\texttt{function(x)}$ returns \texttt{U}.
+    \item \texttt{x} $\in$ \texttt{[L, U]}: then $\texttt{function(x)}$ returns \texttt{x}.
+    \item \texttt{x} $<$ \texttt{L}: then $\texttt{function(x)}$ returns \texttt{L}.
+\end{enumerate}
+In all three cases, the returned value of type \T is contained in the interval \texttt{[L, U]}. Hence, the element $\function(x)$ returned in line~\ref{line:return} of the pseudocode is an element of \texttt{output\_domain}.
+
+Lastly, the necessary condition that \texttt{L} $\leq$ \texttt{U} is checked when declaring \texttt{output\_domain = IntervalDomain(L, U)} in line~\ref{line:output} of the pseudocode. This check already exists via the construction of \texttt{IntervalDomain}, which returns an error if \texttt{L} $>$ \texttt{U}. Both \texttt{L} and \texttt{U} have type \texttt{T} by their precondition requirement. Both the definition of \texttt{IntervalDomain} and that of \texttt{function} (line~\ref{line:fn} in the pseudocode, which uses the \texttt{min} and \texttt{max} functions) require that \texttt{T} implements \texttt{TotalOrd}, which holds by the preconditions.
+
+\smallskip
+\textbf{(Domain-metric compatibility).}  For \texttt{MakeClampAbs}, this corresponds to showing that both \texttt{AllDomain(T)} and \texttt{IntervalDomain(L, U)} are compatible with absolute distance. The first one follows directly from the definition of absolute distance, as stated in \href{https://www.overleaf.com/project/60d214e390b337703d200982}{``List of definitions used in the proofs"}. The second one follows from the compatibility of absolute distance and \texttt{AllDomain(T)} along with the fact that \texttt{IntervalDomain(L, U)} is a subdomain of \texttt{AllDomain(T)}. By Theorem 2.1 in \href{https://www.overleaf.com/project/60d215bf90b337ac02200a99}{``List of definitions used in the pseudocode"}, this implies that \texttt{IntervalDomain(L, U)} is compatible with absolute distance as well. 
+    
+\silvia{Flag: this is an example of the subdomain issues that we have been discussing during the week of July 19. Hence this paragraph might need some phrasing updates when the compatibility pairing constructor and the subdomain trait are implemented.}
+
+\smallskip
+\textbf{(Stability guarantee).} Throughout the stability guarantee proof, we can assume that $\function(x)$ and $\function(y)$ are in the correct output domain, by the \textit{appropriate output domain property} shown above. 
```

Comment:
I don't see how this shows that the stability relation holds. I think your cases are as follows:
1. If the clamping range is less than d_in
2. If the clamping range is greater than d_in

The relation you've shown `d_in <= d_out` doesn't tighten the sensitivity when the clamping range is less than the input distance.

Reply:
Sorry, I didn't tighten the third case. It is fixed now.